### PR TITLE
add more item categories

### PR DESCRIPTION
**Pull request by LordMZTE**
adds more options to the categorized item target

### CODE REVIEW

File: src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt [hunk 1]
```diff
@@ -174,6 +197,65 @@ enum class ItemCategory(override val belongsFunc: (Item) -> Boolean) : Categoris
...
+            // Items.CHARCOAL_BLOCK, oh wait that doesnt exist
+            Items.LAPIS_BLOCK,
+
+            // nuggets
+            Items.IRON_NUGGET,
+            Items.GOLD_NUGGET,
+            // Items.NETHERITE_NUGGET, oh wait that doesnt exist either
+        )
+
+        private val mobDrops = setOf(
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:229]: This is by far not an exhaustive list, but imo it doesn't really need to be - some items are difficult to categorise as 'specifically a mob drop'.

However, I would like to see added:
* Ender pearl
* Gold nugget
* Raw foods: raw beef, raw chicken, raw cod, raw salmon, raw porkchop, raw rabbit, raw mutton

File: src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt [hunk 2]
```diff
@@ -135,15 +139,34 @@ enum class ItemCategory(override val belongsFunc: (Item) -> Boolean) : Categoris
...
+        food.foodComponent?.statusEffects?.stream()
+            ?.allMatch { effect -> effect.first.effectType.isBeneficial }
+            ?: false
+    }),
+
     REDSTONE({ it.group == ItemGroup.REDSTONE || it in additionalRedstone }),
+    RESOURCES({ it in resources }),
+    MOB_DROPS({ it in mobDrops }),
+    ORES({ (it as? BlockItem)?.block is OreBlock || it == Items.ANCIENT_DEBRIS }),
+    WOOD({ (it as? BlockItem)?.block?.defaultState?.material in woodMaterials || it in additionalWood }),
```
Comment (zeroeightysix) [inline src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:168]: What about things like wooden doors, trapdoors, maybe boats, maybe bookshelves, maybe crafting tables, fences, slabs, ...?

Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:168]: all except the boat should be covered by the material check

also im not quite sure if boats would be a good idea. i mean stuff like wooden tools arent included either

Comment (zeroeightysix) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:168]: Good point. If the rest is covered by the material check, I'm happy.

File: src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt (hunk 1 above)
Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:229]: do you think i should remove slime balls from the list? for some reason my brain does not categorize them as mob drop but as redstone crafting componenent

Comment (zeroeightysix) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:229]: I'd prefer for slime balls to stay.

Comment (fee1-dead) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:229]: So you think slime balls are for redstone and redstone is a mob drop?

File: src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt [hunk 3]
```diff
@@ -174,6 +197,65 @@ enum class ItemCategory(override val belongsFunc: (Item) -> Boolean) : Categoris
...
+            Items.SLIME_BALL,
+            Items.SHULKER_SHELL,
+            Items.ARROW,
+            Items.BOW,
+            Items.CROSSBOW,
+
+            // these are dropped by witches
+            Items.GLASS_BOTTLE,
+            Items.REDSTONE,
+            Items.GLOWSTONE_DUST
```
Comment (fee1-dead) [inline src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:257]: I think magma cream dropped from magma cubes should be in this.

File: src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt (hunk 1 above)
Comment (LordMZTE) [reply src/main/kotlin/me/zeroeightsix/kami/target/CategorisedTargets.kt:229]: > 
> 
> So you think slime balls are for redstone and redstone is a mob drop?

its just a subjective thing. ill leave them in the list

**Review by zeroeightysix — APPROVED** [review]

**Comment by zeroeightysix** [comment]
Thanks!

**Comment by humboldt123** [comment]
Your graph will be dark green for today.

**Comment by LordMZTE** [comment]
> 
> 
> Your graph will be dark green for today.

i didnt tell you that im working on the next feature already right?

**Pull request merged**